### DISCUSSION TITLE
Discord as the new Community place

### DISCUSSION THREAD

**Discussion by dadrus** [dadrus/heimdall#591] [Announcements]
As the title implies, I would like to try out a new place for (not yet existing) community around heimdall. 
So, feel free to join https://discord.gg/qQgg8xKuyb.

**Comment by cloudcompute** [comment]
@dadrus

Approximately by when it can be used in production?

Thanks

**Reply by dadrus** [reply]
It is already used in production by multiple projects around the world (according to my knowledge - Canada, Germany, Sweden, Slovenia). You are also very welcome to join the community in Discord and ask about the experience the people there have)

If your question is related to the alpha prefix the releases of the project have. This is due to the amount of the breaking changes I still expect to happen in the configuration, respectively rules. I hope, this will be pretty much reduced after the next release. Nevertheless, there is always a clear migration path available, which you can follow even now without risking downtimes.

**Reply by dadrus** [reply]
When I remove the -alpha prefix, the project will enter the beta phase. That will mean, I will support multiple (at least two) configuration/rule versions in the service itself and you'll get guaranteed dependencies and security updates for at least two versions. Right now it is just the last released minor version,  which is guaranteed to have patch releases at least once in a month

**Reply by dadrus** [reply]
What are your particular concerns? Do you maybe miss something?

**Reply by cloudcompute** [reply]
I do not know much about the open source projects that are popular Go. After some research, I got to know that there are very good IAM, authentication and authorization solutions. I liked Zitadel, OpenFGA and OPA in particular. 

While browsing Github, I had come across this `heimdall` library. Yes, you are right.. the `alpha` prefix is what made me wonder whether the library is still in alpha stagee. thank you for the clarification.

Can I use it for integrating Zitadel with Open FGA / OPA? Hope it will take care of the session. This is my main concern.


PS: I have joined the Discord channel.

**Reply by cloudcompute** [reply]
@dadrus

I have another question. If you can answer it, it'd be great!

If we want to develop a B2B application, we typically need a multi-tenant IAM solution like Keycloak Or Zitadel where we can define Organizations / Realms and their users with passwords etc. 

There are few others like ORY Kratos and Authelia which are not multi-tenant. Is there any workaround to turn such solutions to multi-tenant, for example, adding Organization and other details as metadata?

**Reply by dadrus** [reply]
Sorry for the belated answer.

> Can I use it for integrating Zitadel with Open FGA / OPA?

Yes. Zitadel is "just" an OIDC provider. The integration would happen on another level

> There are few others like ORY Kratos and Authelia which are not multi-tenant. Is there any workaround to turn such solutions to multi-tenant, for example, adding Organization and other details as metadata?

Kratos - definitely. You can do this
Authelia - never worked with it. So, cannot answer your question.

**Comment by cloudcompute** [comment]
thanks for the reply @dadrus 

I was trying to set up Heimdall using the [envoy-proxy](https://github.com/dadrus/heimdall/blob/main/examples/docker-compose/quickstarts/docker-compose-envoy-grpc.yaml).

I had a look at [this file](https://github.com/dadrus/heimdall/blob/main/examples/docker-compose/quickstarts/docker-compose.yaml)

`idp:
    image: nginx:1.25.4
    ...`
    
So the docker-compse tries to install nginx. But I want Envoy instead. Is there any readymade script that can install all 3: Identity provider, OpenFGA or OPA, and any proxy?

**Reply by dadrus** [reply]
Take a look here: https://github.com/dadrus/heimdall/blob/main/examples/docker-compose/quickstarts/README.md
It does however just emulate a specific part of an idp. You can easily extend it with an idp of your choice

**Reply by dadrus** [reply]
May I also ask you for a favor? If you start a new topic, could you do that in a "new discussion". Otherwise, other people won't find anything

**Reply by cloudcompute** [reply]
Ok I will. Thanks.
